marathon_acme/tests/test_cli.py

```diff
@@ -1,27 +1,42 @@
-import re
-
-from click.testing import CliRunner
+from fixtures import TempDir
+from testtools import ExpectedException, run_test_with, TestCase
+from testtools.matchers import Equals, MatchesStructure
+from testtools.twistedsupport import (
+    AsynchronousDeferredRunTest, flush_logged_errors)
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.error import DNSLookupError
+from txacme.urls import LETSENCRYPT_STAGING_DIRECTORY
 
 from marathon_acme.cli import main
 
 
-def test_email_required():
-    """
-    The program is expected to exit with an error code and message if '--email'
-    is not provided.
-    """
-    runner = CliRunner()
-    result = runner.invoke(main, [])
-
-    assert result.exit_code == 2
-    assert re.search('Error: Missing option "--email"', result.output)
-
-
-def test_email_provided():
-    """
-    The program is expected to exit with code 0 if '--email' is provided.
-    """
-    runner = CliRunner()
-    result = runner.invoke(main, ['--email', 'test@example.com'])
-
-    assert result.exit_code == 0
+class TestCli(TestCase):
+    # These are testtools-style tests so we can run aynchronous tests
+
+    def test_storage_dir_required(self):
+        """
+        When the program is run with no arguments, it should exit with code 2
+        because there is one required argument.
+        """
+        with ExpectedException(SystemExit, MatchesStructure(code=Equals(2))):
+            main(reactor, raw_args=[])
+
+    @inlineCallbacks
+    @run_test_with(AsynchronousDeferredRunTest.make_factory(timeout=10.0))
+    def test_storage_dir_provided(self):
+        """
+        When the program is run with an argument, it should start up and run.
+        The program is expected to fail because it is unable to look up the
+        Marathon DNS address.
```

Comment:
I don't want to rely on `marathon.mesos` not being resolvable for this test. Can we maybe use `localhost:20202` or something to get a connection refused or use some domain name we're reasonably sure will never exist?